cleverhans/attacks.py

```diff
@@ -485,6 +494,12 @@ def parse_params(self,
         :param clip_min: (optional float) Minimum input component value
         :param clip_max: (optional float) Maximum input component value
         """
+        if self.child is not None:
+            if self.child == 'BasicIterativeMethod':
+                rand_minmax = 0.
+            elif self.child == 'MadryEtAl':
+                if 'rand_init' in kwargs and not kwargs['rand_init']:
+                    rand_minmax = 0.
```

Comment:
I would suggest the following. Change this line to

    rand_minmax = self.default_rand_minmax()

Then in `BasicIterativeMethod` implement

    def default_rand_minmax(self):
        return 0.

and in the Madry subclass implement

     def default_rand_minmax(self):
          if self.rand_init:
              return self.epsilon
           return 0.

(if epsilon will already have been parsed? maybe epsilon needs to be an argument to this method)